### PR TITLE
[connector/datadog] Add a config bucket_interval

### PR DESCRIPTION
**Pull request by songy23**
`traces::bucket_interval` specifies the time interval size of aggregation buckets that aggregate the Datadog trace metrics. It is also the time interval that Datadog trace metrics payloads are flushed to the pipeline. Default is 10s if unset.

### CODE REVIEW

File: .chloggen/dd-conn-bkt-size.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: datadogconnector
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Add a config `bucket_interval`"
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (songy23) [inline .chloggen/dd-conn-bkt-size.yaml:13]:
```suggestion
issues: [34554]
```

File: connector/datadogconnector/config.go [hunk 2]
```diff
@@ -75,6 +76,11 @@ type TracesConfig struct {
 
 	// ResourceAttributesAsContainerTags specifies the list of resource attributes to be used as container tags.
 	ResourceAttributesAsContainerTags []string `mapstructure:"resource_attributes_as_container_tags"`
+
+	// BucketInterval specifies the time interval size of aggregation buckets that aggregate the Datadog trace metrics.
+	// It is also the time interval that Datadog trace metrics payloads are flushed to the pipeline.
+	// Default is 10s if unset.
```
Comment (dineshg13) [inline connector/datadogconnector/config.go:82]: where is the default set ? i don't see it in the PR

File: .chloggen/dd-conn-bkt-size.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: datadogconnector
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Add a config `bucket_interval`"
```
Comment (mx-psi) [inline .chloggen/dd-conn-bkt-size.yaml:10]: Should this be `traces::bucket_interval`?
```suggestion
note: "Add a config `traces::bucket_interval`"
```

File: .chloggen/dd-conn-bkt-size.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [34554]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: "`bucket_interval` specifies the time interval size of aggregation buckets that aggregate the Datadog trace metrics. It is also the time interval that Datadog trace metrics payloads are flushed to the pipeline. Default is 10s if unset."
```
Comment (mx-psi) [inline .chloggen/dd-conn-bkt-size.yaml:18]:
```suggestion
subtext: "`traces::bucket_interval` specifies the time interval size of aggregation buckets that aggregate the Datadog trace metrics. It is also the time interval that Datadog trace metrics payloads are flushed to the pipeline. Default is 10s if unset."
```

**Review by mx-psi — COMMENTED** [review]
Can you add some brief explanation to the PR description?

File: connector/datadogconnector/config.go (hunk 2 above)
Comment (songy23) [reply connector/datadogconnector/config.go:82]: Default is set by trace agent config: https://github.com/DataDog/datadog-agent/blob/7f37c8d76776d7a5267946683e6931a3a177f68a/pkg/trace/config/config.go#L477

**Comment by songy23** [comment]
> Can you add some brief explanation to the PR description?

Done

**Review by mx-psi — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**